test/oauth-authorization-test.js

```diff
@@ -383,24 +416,37 @@ suite.addBatch({
                         requestToken(cl, this);
                     },
                     function(err, rt) {
+                        var self = this;
+
                         if (err) throw err;
-                        br = new Browser({runScripts: false});
-                        br.visit("http://localhost:4815/oauth/authorize?oauth_token=" + rt.token, this);
+                            br = new Browser({runScripts: false});
+                            Step(
+                                function(){
+                                    br.visit("http://localhost:4815/oauth/authorize?oauth_token=" + rt.token, this);
+                                },
+                                function(){
+                                    self(!br.success,br);
+                                }
+                            );
                     },
                     function(err, br) {
+                        var self = this;
+                        
                         if (err) throw err;
-                        if (!br.success) throw new Error("Browser error");
-                        br.fill("username", "nonexistent", this);
-                    },
-                    function(err) {
-                        if (err) throw err;
-                        br.fill("password", "DOESNTMATTER", this);
-                    },
-                    function(err) {
-                        if (err) throw err;
-                        br.pressButton("#authenticate", this);
+                        br.assert.success();
+                        br
+                            .fill("username", "nonexistent", this)
```

Comment:
Put this first `.fill()` call on the same line as `br` - improves readability IMHO